src/Compilers/CSharp/Portable/Syntax/CSharpPragmaWarningStateMap.cs

```diff
@@ -112,91 +112,71 @@ private static WarningStateMapEntry[] CreatePragmaWarningStateEntries(ArrayBuild
                 {
                     var currentPragmaDirective = (PragmaWarningDirectiveTriviaSyntax)currentDirective;
 
-                    // Compute the directive state (either Disable or Restore)
-                    var directiveState = currentPragmaDirective.DisableOrRestoreKeyword.Kind() == SyntaxKind.DisableKeyword ? PragmaWarningState.Disabled : PragmaWarningState.Default;
-
-                    // Check if this directive applies for all (e.g., #pragma warning disable)
-                    if (currentPragmaDirective.ErrorCodes.Count == 0)
+                    if (currentPragmaDirective.NullableKeyword.IsKind(SyntaxKind.NullableKeyword))
                     {
-                        // Update the warning state and reset the specific one
-                        accumulatedGeneralWarningState = directiveState;
-                        accumulatedSpecificWarningState = ImmutableDictionary.Create<string, PragmaWarningState>();
+                        accumulatedNullableWarningState(currentPragmaDirective.DisableOrRestoreKeyword);
                     }
                     else
                     {
-                        // Compute warning numbers from the current directive's codes
-                        for (int x = 0; x < currentPragmaDirective.ErrorCodes.Count; x++)
-                        {
-                            var currentErrorCode = currentPragmaDirective.ErrorCodes[x];
-                            if (currentErrorCode.IsMissing || currentErrorCode.ContainsDiagnostics)
-                                continue;
+                        // Compute the directive state
+                        PragmaWarningState directiveState;
 
-                            var errorId = string.Empty;
-                            if (currentErrorCode.Kind() == SyntaxKind.NumericLiteralExpression)
-                            {
-                                var token = ((LiteralExpressionSyntax)currentErrorCode).Token;
-                                errorId = MessageProvider.Instance.GetIdForErrorCode((int)token.Value);
-                            }
-                            else if (currentErrorCode.Kind() == SyntaxKind.IdentifierName)
-                            {
-                                errorId = ((IdentifierNameSyntax)currentErrorCode).Identifier.ValueText;
-                            }
+                        switch (currentPragmaDirective.DisableOrRestoreKeyword.Kind())
+                        {
+                            case SyntaxKind.DisableKeyword:
+                                directiveState = PragmaWarningState.Disabled;
+                                break;
+                            case SyntaxKind.RestoreKeyword:
+                                directiveState = PragmaWarningState.Default;
```

Comment:
Allowing `restore` for error codes too is not covered in [https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md](https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md)
Please confirm with Mads and get it added to the spec if appropriate.

Reply:
> Allowing restore for error codes too is not covered in [https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md](https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md)
Please confirm with Mads and get it added to the spec if appropriate.

There is no change around ```restore```, it was always supported. The added actions are ```enable``` and ```safeonly```.


---
In reply to: [245837409](https://github.com/dotnet/roslyn/pull/32208#discussion_r245837409) [](ancestors = 245837409)

Reply:
My bad, I thought we previously supported `enable` and `disable`.
Then my question should be on `enable`. What happens with `#pragma warning enable 1234`?

---
In reply to: [245838133](https://github.com/dotnet/roslyn/pull/32208#discussion_r245838133) [](ancestors = 245838133,245837409)

Reply:
> Then my question should be on enable. What happens with #pragma warning enable 1234?

It enables the warning, this is also part of the spec 

---
In reply to: [245839503](https://github.com/dotnet/roslyn/pull/32208#discussion_r245839503) [](ancestors = 245839503,245838133,245837409)